docs/api/url_search_params.md

```diff
@@ -0,0 +1,99 @@
+# 🔗 URLSearchParams
+
+`jstz` implementation of an API with utility methods for working with the query string of a URL according to the [URLSearchParams specification](https://url.spec.whatwg.org/#urlsearchparams). It is used for building and manipulating search parameters.
+
+## Example
+
+```typescript
+// Parse query string from URL
+let url = new URL("https://example.com?foo=1&bar=2");
+let params = new UrlSearchParams(url.search);
+
+// Add a new parameter
+params.append("baz", 3);
+
+// Remove parameter
+params.delete("bar");
+
+// Loop through parameters
+for (let [key, value] of params) {
+  // ...
+}
+```
+
+## Constructor
+
+### `UrlSearchParams(values: Array<[Name: string, Value: string]>, url?: URL)`
+
+Creates a new instance of `UrlSearchParams` with the provided key-value pairs.
+
+- **values**: An array of key-value pairs. Each pair is an array where the first element is the key (Name) and the second is the value.
+
+## Instance Methods
+
+### `setValues(values: Array<[Name: string, Value: string]>): void`
+
+Sets the current values with the provided array of key-value pairs.
+
+- **values**: An array of key-value pairs. Each pair consists of a name (key) and value.
+
+### `setUrl(url: JsNativeObject<Url>): void`
+
+Associates the `UrlSearchParams` with a given `Url` object.
+
+- **url**: A reference to the `Url` object.
+
+### `len(): number`
+
+- **Returns**: The number of search parameters present.
+
+### `append(name: string, value: string): void`
+
+Appends a specified key/value pair as a new search parameter.
+
+- **name**: The name of the search parameter.
+- **value**: The value of the search parameter.
```

Comment:
We don't use this style of specifying arguments in our documentation. I'm not opposed to it, but it is inconsistent with what the rest of the docs does